MVC-App/Services/RelationService.cs

```diff
@@ -0,0 +1,280 @@
+{
+    public class RelationService : IRelationService
+    {
+        private readonly UnitOfWork _unitOfWork;
+
+        public SelectList Categories { get; set; }
+
+        public SelectList Countries { get; set; }
+
+        public RelationService(UnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+
+            var categories = unitOfWork.CategoryRepository.Get().ToList();
+
+            categories.Insert(0, new Category
+            {
+                Id = Guid.Empty,
+                CreatedAt = DateTime.Now,
+                CreatedBy = "admin",
+                IsDisabled = false,
+                Name = "Все"
+            });
+
+            Categories = new SelectList(categories, "Id", "Name");
+
+            var countries = unitOfWork.CountryRepository.Get().ToList();
+
+            Countries = new SelectList(countries, "Id", "Name");
+        }
+
+        public async Task<List<RelationVM>> InitRelationModels()
+        {
+            
+
+            var relationModels = from relation in _unitOfWork.RelationRepository.Get()
+                                 join relationAddress in _unitOfWork.RelationAddressRepository.Get() on relation.Id equals relationAddress.RelationId
+                                 join country in _unitOfWork.CountryRepository.Get() on relationAddress.CountryId equals country.Id
+                                 where !relation.IsDisabled
+                                 select new RelationVM
+                                 {
+                                     Id = relation.Id,
+                                     RelationAddressId = relationAddress.Id,
+                                     Categories = (from relationCategory in _unitOfWork.RelationCategoryRepository.Get() where relationCategory.RelationId == relation.Id select relationCategory.CategoryId).ToList(),
+                                     Name = relation.Name,
+                                     FullName = relation.FullName,
+                                     TelephoneNumber = relation.TelephoneNumber,
+                                     Email = relation.EMailAddress,
+                                     CountryId = country.Id,
+                                     CountryName = country.Name,
+                                     City = relationAddress.City,
+                                     Street = relationAddress.Street,
+                                     PostalCode = relationAddress.PostalCode,
+                                     PostalCodeMask = country.PostalCodeFormat,
+                                     StreetNumber = relationAddress.Number ?? 0
+                                 };
+            return relationModels.ToList();
+        }
+
+        public string ApplyMask(string value, string mask)
+        {
+            if (mask == string.Empty || mask == null || value == string.Empty) return value;
+
+            List<char> result = new List<char>();
+
+            int valueIterator = 0;
```

Comment:
can you make:
int valueIterator, valueNumbersCount, maskNumbersCount, maskLettersCount = 0;